18F/e-QIP-prototype
# [EN-3845] Create SF85 test scenarios

## Description

1. This PR creates three SF85 specific test scenarios.
2. This PR makes removes unnecessary validations from the SF86 that aren't a part of the SF85.
3. This PR makes updates to the `load-scenario` tool to allow the SF85 test scenarios to be loaded into eApp.

## # 1
`test1.json` is the simplest scenario of the full SF85 form going down as few branched sections as possible. 
`test2.json`is a complete blowout of the full SF85 form going down as many branched sections as possible.
`test3.json`is a complete blowout of the full SF85 form, similar to `test2.json`
*It is also worth mentioning `test2.json` and `test3.json` were created by two different people.

A new folder has also been created in to hold the new SF85 test scenarios: `api/testdata/complete-scenarios/SF85`

### How do I load these tests cases?
1. Make sure to run `make build-cmd`
2. Use `docker exec CONTAINERID sftype USERNAME SF_TYPE SF_VERSION` to update a user to have the SF85 form.
`SF_TYPE` should be `SF85`
`SF_VERSION` should be `2017-12-draft7`
3. After that, you should be set to follow the instructions [here](https://github.com/18F/e-QIP-prototype/blob/develop/docs/test-scenarios.md#loading-existing-test-json-files)

## # 2
Because the SF85 is a subset of the SF86, there are many fields/validations that needed to be removed. During that process, a few validations were overlooked. Those validations prevent the form from being fully validated because those fields aren't on the SF85 to be filled out. Test cases 2 and 3 in the PR will not pass without these changes. This worked is tracked by `EN-3856`. A summary of the issues I found are listed below:

| Section | Form Tracking Row |
| -------- | ------------------- | 
| Substance Use / Illegal Use of Drugs and Drug Activity / Usage | 680 |
| Investigative and criminal history / Police record / Offenses |  624-626 |
| Investigative and criminal history / Police record / Offenses | 639-640 |

## # 3
The `load-scenario` tool is reading from a catalog of sections/subsections. If any of those sections/subsections are missing from the JSON getting imported, it throws a fatal error. To resolve that, it uses `Printf()` instead `Fatalf()`.

## Notes
I think this is outside of the scope of the PR, but there is extraneous data in the SF85 test scenarios JSON. Unnecessary sections are properly excluded from the JSON, but subsections are not excluded from the SF85 JSON (they're empty). Please review the JSON files for further understanding.

~While filling out the form, the Redux store does not populate those subsections, nor are they sent in the payload when saving. At some point in the backend, it is creating those empty data structures. When logging back in, those empty data structures show up in Redux even though they weren't there previously.~

If this doesn't make complete sense, I can write a more thorough issue with pictures and GIFs.


## Checklist for Reviewer

- [ ] Review code changes
- [ ] Review changes for Database effects
- [ ] Verify change works in IE browser

More details about this can be found in [docs/review.md](docs/review.md)
